src/cache/cloud_fraction.jl

```diff
@@ -272,3 +285,89 @@ NVTX.@annotate function set_cloud_fraction!(
         end
     end
 end
+
+function set_ml_cloud_fraction(
+    Y,
+    p,
+    cloud_ml::CloudML,
+    thermo_params,
+    turbconv_model,
+    ᶜts,
+)
+    FT = eltype(p.params)
+    ᶜmixing_length_field = p.scratch.ᶜtemp_scalar
+    ᶜmixing_length_field .=
+        turbconv_model isa PrognosticEDMFX || turbconv_model isa DiagnosticEDMFX ?
+        ᶜmixing_length(Y, p) :
+        compute_gm_mixing_length(Y, p)
+
+    # Vertical gradients of q_tot and θ_liq_ice
+    ᶜ∇q = p.scratch.ᶜtemp_scalar_2
+    ᶜ∇q .=
+        projected_vector_data.(
+            C3,
+            p.precomputed.ᶜgradᵥ_q_tot,
+            Fields.level(Fields.local_geometry_field(Y.c)),
+        )
+    ᶜ∇θ = p.scratch.ᶜtemp_scalar_3
+    ᶜ∇θ .=
+        projected_vector_data.(
+            C3,
+            p.precomputed.ᶜgradᵥ_θ_liq_ice,
+            Fields.level(Fields.local_geometry_field(Y.c)),
+        )
+
+    q_v = p.scratch.ᶜtemp_scalar_4
+    q_v .= specific.(Y.c.ρq_tot, Y.c.ρ)
+    # Saturation state at current thermodynamic state
+    q_sat = p.scratch.ᶜtemp_scalar_5
```

Comment:
ᶜtemp_scalar_5 is used in ᶜmixing_length called on line 301. It is materialized, so it is probably ok to reuse it here, but it would be good to make sure. An easy way to check is to change the temp_scalar here and see if the result is the same.